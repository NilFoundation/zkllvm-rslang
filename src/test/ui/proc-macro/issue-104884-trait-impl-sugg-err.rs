// aux-build:issue-104884.rs

use std::collections::BinaryHeap;

#[macro_use]
extern crate issue_104884;

#[derive(PartialEq, Eq, PartialOrd, Ord)]
struct PriorityQueueEntry<T> {
    value: T,
}

#[derive(PartialOrd, AddImpl)]
//~^ ERROR can't compare `PriorityQueue<T>` with `PriorityQueue<T>`
//~| ERROR the trait bound `PriorityQueue<T>: Eq` is not satisfied
//~| ERROR can't compare `T` with `T`

struct PriorityQueue<T>(BinaryHeap<PriorityQueueEntry<T>>);

fn main() {}
