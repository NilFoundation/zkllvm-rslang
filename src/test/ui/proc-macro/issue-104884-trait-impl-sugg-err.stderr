error[E0277]: can't compare `PriorityQueue<T>` with `PriorityQueue<T>`
  --> $DIR/issue-104884-trait-impl-sugg-err.rs:13:10
   |
LL | #[derive(PartialOrd, AddImpl)]
   |          ^^^^^^^^^^ no implementation for `PriorityQueue<T> == PriorityQueue<T>`
   |
   = help: the trait `PartialEq` is not implemented for `PriorityQueue<T>`
note: required by a bound in `PartialOrd`
  --> $SRC_DIR/core/src/cmp.rs:LL:COL
   |
LL | pub trait PartialOrd<Rhs: ?Sized = Self>: PartialEq<Rhs> {
   |                                           ^^^^^^^^^^^^^^ required by this bound in `PartialOrd`
   = note: this error originates in the derive macro `PartialOrd` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `PriorityQueue<T>: Eq` is not satisfied
  --> $DIR/issue-104884-trait-impl-sugg-err.rs:13:22
   |
LL | #[derive(PartialOrd, AddImpl)]
   |                      ^^^^^^^ the trait `Eq` is not implemented for `PriorityQueue<T>`
   |
note: required by a bound in `Ord`
  --> $SRC_DIR/core/src/cmp.rs:LL:COL
   |
LL | pub trait Ord: Eq + PartialOrd<Self> {
   |                ^^ required by this bound in `Ord`
   = note: this error originates in the derive macro `AddImpl` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: can't compare `T` with `T`
  --> $DIR/issue-104884-trait-impl-sugg-err.rs:13:22
   |
LL | #[derive(PartialOrd, AddImpl)]
   |                      ^^^^^^^ no implementation for `T < T` and `T > T`
   |
note: required for `PriorityQueue<T>` to implement `PartialOrd`
  --> $DIR/issue-104884-trait-impl-sugg-err.rs:13:10
   |
LL | #[derive(PartialOrd, AddImpl)]
   |          ^^^^^^^^^^
note: required by a bound in `Ord`
  --> $SRC_DIR/core/src/cmp.rs:LL:COL
   |
LL | pub trait Ord: Eq + PartialOrd<Self> {
   |                     ^^^^^^^^^^^^^^^^ required by this bound in `Ord`
   = note: this error originates in the derive macro `AddImpl` which comes from the expansion of the derive macro `PartialOrd` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0277`.
